packages/element/src/create-interpolate-element.js

```diff
@@ -0,0 +1,335 @@
+/**
+ * External dependencies
+ */
+import { createElement, cloneElement, Fragment, isValidElement } from 'react';
+
+let indoc,
+	offset,
+	output,
+	stack;
+
+/**
+ * Matches tags in the localized string
+ *
+ * This is used for extracting the tag pattern groups for parsing the localized
+ * string and along with the map converting it to a react element.
+ *
+ * There are four references extracted using this tokenizer:
+ *
+ * match: Full match of the tag (i.e. <strong>, </strong>, <br/>)
+ * isClosing: The closing slash, it it exists.
+ * name: The name portion of the tag (strong, br) (if )
+ * isSelfClosed: The slash on a self closing tag, if it exists.
+ *
+ * @type RegExp
+ */
+const tokenizer = /<(\/)?(\w+)\s*(\/)?>/g;
+
+/**
+ * This receives the value for a map element which consists of an element and
+ * it's props and returns a element creator function.
```

Comment:
Looks like this is explaining _what_ is occurring but I'm confused _why_ it's happening. Maybe we can indicate the role of this function more specifically and give people more confidence to know what they can and can't change without breaking any assumptions.

Also, do we need this function at all? We should be able to validate the elements at the start of the function call and abort if any of them are invalid.

Reply:
Are you suggesting doing something like:

```
if ( ! Object.values( conversionMap ).some( a => isValidElement( a ) ) {
  throw Error();
}
```

Reply:
> Also, do we need this function at all?

Assuming the above is what you meant (and do that in the `isValidConversionMap` function) then ya we wouldn't need this function at all we could just call `cloneElement` directly where needed.

Reply:
Dropped `elementCreator` and refactored to more directly implement cloneElement and do conversionMap validation early in 0c6c0ae.